### PR TITLE
Ubiquiti devices

### PR DESCRIPTION
**Pull request by tfl-jb**
I updated all Ubitquiti devices that I was able to get my hands on. I added a few new ones and on existing ones I added the proper PoE options for the interfaces.

I ran the pre-commit tests successfully but have to say it is a pain doing that on Windows 😅

### CODE REVIEW

**Comment by tfl-jb** [comment]
Is there any thing I can do to not get this PR deleted?

**Comment by stereu** [comment]
Great commit! Sad to see this PR and others becoming stale due to no activity from the repo maintainer. Would be great if he'll add some more admins here.

**Comment by danner26** [comment]
@tfl-jb @stereu, they have been very busy but I am here to help! Since there are a decent amount of changes let me review them and get back to you

File: device-types/Ubiquiti/USW-Pro-48.yaml
Comment (danner26) [inline device-types/Ubiquiti/USW-Pro-48.yaml:8]: Please remove the newline here

File: device-types/Ubiquiti/USP-RPS.yaml
Comment (danner26) [inline device-types/Ubiquiti/USP-RPS.yaml:9]: Please remove the newline

File: device-types/Ubiquiti/USP-RPS.yaml
Comment (danner26) [inline device-types/Ubiquiti/USP-RPS.yaml:11]: Please remove the newline

File: device-types/Ubiquiti/UDM-Pro.yaml
Comment (danner26) [inline device-types/Ubiquiti/UDM-Pro.yaml:3]: `UniFi` should be included in the begining as it is the product name. The manufacturer is Ubiquiti and the product name is the `UniFi Dream Machine Pro`

File: device-types/Ubiquiti/UCK-G2.yml
Comment (danner26) [inline device-types/Ubiquiti/UCK-G2.yml:5]: Please add `u_height: 0`

File: device-types/Ubiquiti/UCK-G2-PLUS.yml
Comment (danner26) [inline device-types/Ubiquiti/UCK-G2-PLUS.yml:5]: Please add `u_height: 0`

**Review by danner26 — CHANGES_REQUESTED** [review]
Added requests for changes
Reminder, all files should have a trailing newline at the end of them

**Comment by danner26** [comment]
@tfl-jb along with the reviews above, on your new files please make sure you have a trailing newline. The files are:
`device-types/Ubiquiti/USW-Pro-48.yaml` and `device-types/Ubiquiti/USP-RPS.yaml`

Thank you!

File: device-types/Ubiquiti/UCK-G2.yml
Comment (tfl-jb) [reply device-types/Ubiquiti/UCK-G2.yml:5]: It actually has a height of 1 and you can order a [mount kit](https://store.ui.com/collections/unifi-accessories/products/cloud-key-g2-rack-mount-accessory) to fit it in a rack.

**Comment by tfl-jb** [comment]
> @tfl-jb along with the reviews above, on your new files please make sure you have a trailing newline. The files are: `device-types/Ubiquiti/USW-Pro-48.yaml` and `device-types/Ubiquiti/USP-RPS.yaml`
> 
> Thank you!

Thank you for your review @danner26! I am happy to proceed working on this project. Unfortunately the "newline" topic confuses me. When I look at the files in the browser, Github doesn't show the empty line 36 on `device-types/Ubiquiti/USP-RPS.yaml`. Looking at the file in an editor on my machine shows the empty line as well as `CRLF` as line ending. I remember having to fiddle with these options for a while until I passed the pre-commit checks.

What would be the appropriate configuration? You mentioned two device files, but I thought to have used the same settings for all my changed files.

File: device-types/Ubiquiti/UCK-G2.yml
Comment (danner26) [reply device-types/Ubiquiti/UCK-G2.yml:5]: Yup, but the device itself isnt rack mountable without that adapter.. I think we should try to find a better way to model this for the case where you have a UCK-G2 or UCK-G2+ but no rack mount adapter

Maybe others can chime in on this and we can figure out a better option (or just make it a 1U device). Would you like to create a poll in the discussion board or would you like me to?

**Comment by danner26** [comment]
@tfl-jb You know I think I noticed that when helping some others. I wasnt sure if I was miscommunicating something or what, but I kept seeing two new lines. 

The best way to make sure it passes validation is to run `pre-commit run` before you run the `git commit` command. 
Please also merge in the master branches changes before we merge this PR. You can do so in two ways. A) you can sync the upstream master on your fork, then merge your forks master into your branch OR B) you can click the update branch button here, solve any merge conflicts, then do a `git pull` from your local console. From there you can run `pre-commit run -a` which should fix any issues with your new/edited files

If you need any help with this just let me know

File: device-types/Ubiquiti/UCK-G2.yml
Comment (tfl-jb) [reply device-types/Ubiquiti/UCK-G2.yml:5]: If you don't mind, please setup the poll 👍

Comment (danner26) [reply device-types/Ubiquiti/UCK-G2.yml:5]: The more I thought it over in my head the more complicated it became. Im retracting my previous opinion
Lets keep it with u_height 1 for now, and if someone comes up with a better way in the future we can open another PR for that

**Comment by danner26** [comment]
@tfl-jb after merging in the master branch, please check the run details and fix any errors
If you use `pre-commit run` before doing the `git commit` command, it should catch any issues that appear in the GitHub Actions run

File: device-types/Ubiquiti/USP-RPS.yaml
Comment (danner26) [inline device-types/Ubiquiti/USP-RPS.yaml:3]: The model should be: `UniFi SmartPower RPS` as described on the datasheet

File: device-types/Ubiquiti/USP-RPS.yaml
Comment (danner26) [inline device-types/Ubiquiti/USP-RPS.yaml:4]: The slug should reflect the SKU (`USP-RPS`) or the name (`unifi-smartpower-rps`)

**Review by danner26 — CHANGES_REQUESTED** [review]
Please fix these issues as well

**Comment by tfl-jb** [comment]
I updated the branch here without issues as it seems. I also installed `pre-commit` in my WSL environment and it took a minute to check all files and then it said everything seems to be in order. But, to be honest, I am still not sure if I use it right.

When I have a modified file staged and run `pre-commit run` it will tell me that I have to stage `.pre-commit-config.yaml` as well. Once I did that and run the pre-commit command again it will check all files.

```$ pre-commit run
[WARNING] Unstaged files detected.
[INFO] Stashing unstaged files to /some/path/.cache/pre-commit/patch1673003775-1775.
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
Check YAML files.........................................................Passed
Format YAML files........................................................Passed
Lint YAML files..........................................................Passed
[INFO] Restored changes from /some/path/.cache/pre-commit/patch1673003775-1775.
```

However, now git tells me that there are hundres of changes on all files in my current branch. Comparing the files shows no changes 😕

One example from the `module-type` folder, where I didn't touch anything.
```
diff --git a/module-types/Ubiquiti/RPS-AC-100W.yaml b/module-types/Ubiquiti/RPS-AC-100W.yaml
index b83e6e2..17c061b 100644
--- a/module-types/Ubiquiti/RPS-AC-100W.yaml
+++ b/module-types/Ubiquiti/RPS-AC-100W.yaml
@@ -1,10 +1,10 @@
----
-manufacturer: Ubiquiti
-model: RPS-AC-100W
-part_number: RPS-AC-100W
-comments: |
-  [Ubiquiti RPS-AC-100W](https://dl.ui.com/ds/uf_gpon) Page 5
-power-ports:
-  - name: PSU {module}
-    type: iec-60320-c14
-    maximum_draw: 100
+---
+manufacturer: Ubiquiti
+model: RPS-AC-100W
+part_number: RPS-AC-100W
+comments: |
+  [Ubiquiti RPS-AC-100W](https://dl.ui.com/ds/uf_gpon) Page 5
+power-ports:
+  - name: PSU {module}
+    type: iec-60320-c14
+    maximum_draw: 100
```

Since the `diff` faults the whole file I can only assume that something with line endings is the problem. Maybe I should simply use Githubs built in web editor to work on files for this project 😅

**Comment by danner26** [comment]
Hey @tfl-jb if you want to give me access to your fork, I can cleanup your repo and get it in a working state for pre-commit to work. If you have pulled in the latest master branch you will have the changes we recently made
You should be able to run `pre-commit run -a` which will run on all files, and subsequently it should fix any issues (if they exist)

**Comment by tfl-jb** [comment]
> Hey @tfl-jb if you want to give me access to your fork, I can cleanup your repo and get it in a working state for pre-commit to work. If you have pulled in the latest master branch you will have the changes we recently made You should be able to run `pre-commit run -a` which will run on all files, and subsequently it should fix any issues (if they exist)

Many thanks, I have invited you.

**Review by danner26 — APPROVED** [review]

**Comment by danner26** [comment]
Thanks for all of your work on this @tfl-jb!

**Comment by tfl-jb** [comment]
Awesome!

**Pull request merged**
